app/actions/EmergencyActions.scala

```diff
@@ -0,0 +1,85 @@
+package actions
+
+import config.{AWS, LoginConfig, Off, On}
+import play.api.Logger
+import play.api.mvc.Results._
+import play.api.mvc.{ActionBuilder, Headers, Request, Result}
+import com.github.t3hnar.bcrypt._
+import com.gu.scanamo._
+import com.gu.scanamo.syntax._
+import scala.concurrent.Future
+
+object EmergencySwitchIsOnAction extends ActionBuilder[Request] {
+  override def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]): Future[Result] = {
+    config.Switches.allSwitches.get("emergency") match {
+      case Some(On) => block(request)
+      case Some(Off) => Future.successful(SeeOther("/emergency/reissue-disabled"))
+      case _ => Future.successful(BadRequest("Emergency reissue config switch is not configured correctly, value must be 'on' or 'off'."))
+    }
+  }
+}
+
+object EmergencySwitchChangeAccess extends ActionBuilder[Request] {
+
+  lazy val loginConfig = LoginConfig.loginConfig(AWS.eC2Client)
+
+  override def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]): Future[Result] = {
+
+    def checkPassword(user: EmergencyUser, username: String, password: String): Future[Result] = {
+      if (password.isBcrypted(user.passwordHash)) {
+        block(request)
+      } else {
+        refuseSwitchChange(s"The password provided by ${username} is incorrect. User will be refused access to change emergency switch.")
+      }
+    }
+
+    def refuseSwitchChange(logErrorMsg: String): Future[Result] = {
+      Logger.error(logErrorMsg)
+      Future.successful {
+        Unauthorized(views.html.switches.switchChange("Authorisation checks failed, the Emergency switch will not be changed. Contact digitalcms.dev@theguardian.com for more help."))
+      }
+    }
+
+    try {
+      val authHeaderUser = getBasicAuthDetails(request.headers)
+      val userId = authHeaderUser.id
+      val userOpt = Scanamo.get[EmergencyUser](AWS.dynamoDbClient)(s"login.gutools-emergency-access-${loginConfig.stage.toUpperCase}")('userId -> s"$userId")
+
+      userOpt.map { xor =>
```

Comment:
Would this be better written with `case` in the map so you don't get the nested invocations here?

Something like this, perhaps? (with `import Xor.{Left, Right}`)

    userOpt map {
      case Left(error) => refuseSwitchChange(...)
      case Right(user) => checkPassword(...)
    } getOrElse refuseSwitchChange(...)